news/7731.bugfix

```diff
@@ -0,0 +1 @@
+Avoid listing packages from current directory, when invoked as 'python -m pip list'
```

Comment:
Perhaps mention other known commands affect by this (freeze and check).

Reply:
I think `show` will also be affected by this change.

Reply:
Hi @sbidoul 

I have made the necessary changes, also adding `show` to the list

Reply:
> I think `show` will also be affected by this change.

Hi @pradyunsg 

Could you give me an example of how `show` will be affected here, so I can try it out and confirm the same

Reply:
If a package is present in cwd and you run `pip show foo`, it doesn't show the details of the package instead errors out as `WARNING: Package(s) not found: foo`. However, when you do the same with `python -m pip show foo`, it does show the details.

With this change, the behavior would be same for both ways of pip invocations.

Reply:
Thanks @gutsytechster 

And I am assuming that since we want the same details with both variants, this PR will positively affect `show` and even `install` as @sbidoul mentioned at https://github.com/pypa/pip/pull/7955#issuecomment-609387405